profiling/src/sapi.rs

```diff
@@ -40,6 +44,32 @@ impl Sapi {
             Some(sapi) => *sapi,
         }
     }
+
+    pub fn request_script_name(&self, sapi_globals: &sapi_globals_struct) -> Option<String> {
+        match self {
+            /* Right now all we need is CLI support, but theoretically it can
+             * be obtained for web requests too if we care.
+             */
+            Sapi::Cli => {
+                let request_info = &sapi_globals.request_info;
+                if request_info.argc > 0 && !request_info.argv.is_null() {
+                    // Safety: It's not null; the VM should do the rest.
+                    let cstr = unsafe { CStr::from_ptr(*request_info.argv) };
+                    return if !cstr.to_bytes().is_empty() {
+                        let osstr = OsStr::from_bytes(cstr.to_bytes());
+                        Path::new(osstr)
```

Comment:
Are there any benefits of going through the path object versus just converting to a string slice ?

Reply:
I need only the file name, not the whole path, which we get from `Path::file_name`. Path's need something that is `AsRef<OsStr>` and `&CStr` doesn't count, so I convert to `OsStr` in the middle.